modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/domain/nodeCollateral/UpdateNodeCollateralValidator.scala

```diff
@@ -0,0 +1,203 @@
+package io.constellationnetwork.node.shared.domain.nodeCollateral
+
+import cats.data.{NonEmptySet, ValidatedNec}
+import cats.effect.Async
+import cats.syntax.all._
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+import io.constellationnetwork.ext.cats.syntax.validated._
+import io.constellationnetwork.node.shared.domain.nodeCollateral.UpdateNodeCollateralValidator.UpdateNodeCollateralValidationErrorOr
+import io.constellationnetwork.node.shared.domain.seedlist.SeedlistEntry
+import io.constellationnetwork.schema.address.Address
+import io.constellationnetwork.schema.delegatedStake.UpdateDelegatedStake
+import io.constellationnetwork.schema.epoch.EpochProgress
+import io.constellationnetwork.schema.nodeCollateral._
+import io.constellationnetwork.schema.peer.PeerId
+import io.constellationnetwork.schema.tokenLock.{TokenLock, TokenLockReference}
+import io.constellationnetwork.schema.{GlobalSnapshotInfo, SnapshotOrdinal}
+import io.constellationnetwork.security.signature.SignedValidator.SignedValidationError
+import io.constellationnetwork.security.signature.signature.SignatureProof
+import io.constellationnetwork.security.signature.{Signed, SignedValidator}
+import io.constellationnetwork.security.{Hashed, Hasher, SecurityProvider}
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.types.numeric.NonNegLong
+
+trait UpdateNodeCollateralValidator[F[_]] {
+  def validateCreateNodeCollateral(
+    signed: Signed[UpdateNodeCollateral.Create],
+    lastContext: GlobalSnapshotInfo,
+    lastGlobalEpochProgress: EpochProgress
+  ): F[UpdateNodeCollateralValidationErrorOr[Signed[UpdateNodeCollateral.Create]]]
+  def validateWithdrawNodeCollateral(
+    signed: Signed[UpdateNodeCollateral.Withdraw],
+    lastContext: GlobalSnapshotInfo
+  ): F[UpdateNodeCollateralValidationErrorOr[Signed[UpdateNodeCollateral.Withdraw]]]
+}
+
+// TODO Token lock must have indefinite expiration.
+object UpdateNodeCollateralValidator {
+
+  def make[F[_]: Async: SecurityProvider](
+    signedValidator: SignedValidator[F],
+    seedlist: Option[Set[SeedlistEntry]],
+    minEpochProgressesToLock: NonNegLong
+  )(
+    implicit hasher: Hasher[F]
+  ): UpdateNodeCollateralValidator[F] =
+    new UpdateNodeCollateralValidator[F] {
+
+      def validateCreateNodeCollateral(
+        signed: Signed[UpdateNodeCollateral.Create],
+        lastContext: GlobalSnapshotInfo,
+        lastGlobalEpochProgress: EpochProgress
+      ): F[UpdateNodeCollateralValidationErrorOr[Signed[UpdateNodeCollateral.Create]]] =
+        for {
+          numberOfSignaturesV <- validateNumberOfSignatures(signed)
```

Comment:
Same problem here, we have almost identical validator functions duplicated here.